packages/admin-next/dashboard/src/v2-routes/campaigns/common/components/create-campaign-form-fields/create-campaign-form-fields.tsx

```diff
@@ -0,0 +1,268 @@
+import { useWatch } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import { Form } from "../../../../../components/common/form"
+import { currencies, getCurrencySymbol } from "../../../../../lib/currencies"
+
+export const CreateCampaignFormFields = ({ form, fieldScope = "" }) => {
+  const { t } = useTranslation()
+
+  const watchValueType = useWatch({
+    control: form.control,
+    name: `${fieldScope}budget.type`,
+  })
+
+  const isTypeSpend = watchValueType === "spend"
+
+  const currencyValue = useWatch({
+    control: form.control,
+    name: `${fieldScope}currency`,
+  })
+
+  return (
+    <div className="flex w-full max-w-[720px] flex-col gap-y-8">
+      <div>
+        <Heading>{t("campaigns.create.header")}</Heading>
+
+        <Text size="small" className="text-ui-fg-subtle">
+          {t("campaigns.create.hint")}
+        </Text>
+      </div>
+
+      <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}name`}
+          render={({ field }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("fields.name")}</Form.Label>
+
+                <Form.Control>
+                  <Input {...field} />
+                </Form.Control>
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}description`}
+          render={({ field }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("fields.description")}</Form.Label>
+
+                <Form.Control>
+                  <Input {...field} />
+                </Form.Control>
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}campaign_identifier`}
+          render={({ field }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("campaigns.fields.identifier")}</Form.Label>
+
+                <Form.Control>
+                  <Input {...field} />
+                </Form.Control>
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}currency`}
+          render={({ field: { onChange, ref, ...field } }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("fields.currency")}</Form.Label>
+                <Form.Control>
+                  <Select {...field} onValueChange={onChange}>
+                    <Select.Trigger ref={ref}>
+                      <Select.Value />
+                    </Select.Trigger>
+
+                    <Select.Content>
+                      {Object.values(currencies).map((currency) => (
```

Comment:
question: should we have store currencies here and not all currencies? (if not should we make this a searchable dropdown)

suggestion: also currency formatting is a bit off should we disaply code and smbol as well?

Reply:
Yes, Oli also added this. This will come as a follow up PR.